### PR TITLE
Create faq.md

### PR DESCRIPTION
**Pull request by AttilaTheFun**

### CODE REVIEW

**Comment by 7AXEL** [comment]
Check out this project: https://github.com/7AXEL/Web2APK

File: doc/faq.md [hunk 1]
```diff
@@ -0,0 +1,39 @@
+# Frequently Asked Questions
+
+## How can I build a swift_library directly for iOS?
+
+By default, swift_library will compile code for your host machine, typically macOS.
```
Comment (brentleyjones) [inline doc/faq.md:5]:
```suggestion
By default, `swift_library` will compile code for your host machine (e.g. macOS).
```

File: doc/faq.md [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+As a result, you may see compiler errors if you attempt to build a swift_library
```
Comment (brentleyjones) [inline doc/faq.md:6]:
```suggestion
As a result, you may see compiler errors if you attempt to build a `swift_library`
```

File: doc/faq.md [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+which imports iOS system frameworks like UIKit or SwiftUI.
```
Comment (brentleyjones) [inline doc/faq.md:7]:
```suggestion
which imports iOS system frameworks, such as UIKit or SwiftUI:
```

File: doc/faq.md [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+
+```
+error: no such module 'UIKit'
+```
+
+In order to build a swift_library for iOS, you have two main options:
```
Comment (brentleyjones) [inline doc/faq.md:13]:
```suggestion
In order to build a `swift_library` for iOS, you have two main options:
```

File: doc/faq.md [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+1. (Preferred) Build the swift_library transitively via an ios_ target.
```
Comment (brentleyjones) [inline doc/faq.md:15]:
```suggestion
1. (Preferred) Build the `swift_library` transitively via an `ios_*` target.
```

File: doc/faq.md [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+
+The ios_ rules (ios_application, ios_framework, ios_unit_test, ios_build_test etc)
```
Comment (brentleyjones) [inline doc/faq.md:17]:
```suggestion
The `ios_*` rules (e.g. `ios_application`, `ios_framework`, `ios_unit_test`, `ios_build_test`, etc.) 
```

File: doc/faq.md [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+```
+error: no such module 'UIKit'
+```
+
+In order to build a swift_library for iOS, you have two main options:
+
+1. (Preferred) Build the swift_library transitively via an ios_ target.
+
+The ios_ rules (ios_application, ios_framework, ios_unit_test, ios_build_test etc) 
+from rules_apple apply a "transition" to their transitive dependencies which
```
Comment (brentleyjones) [inline doc/faq.md:18]:
```suggestion
from **rules_apple** apply a "transition" to their transitive dependencies which
```

File: doc/faq.md [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+applies a number of configurations which enable them to build for the target platform.
```
Comment (brentleyjones) [inline doc/faq.md:19]:
```suggestion
applies a number of configurations that enable them to build for the target platform.
```

File: doc/faq.md [hunk 9]
```diff
@@ -0,0 +1,39 @@
...
+
+The benefit to using these rules is that you can share a build cache with your main application,
```
Comment (brentleyjones) [inline doc/faq.md:21]:
```suggestion
The benefit to using these rules is that you can share a analysis and build cache with your main application,
```

File: doc/faq.md [hunk 10]
```diff
@@ -0,0 +1,39 @@
...
+making building targets with a lot of dependencies much faster.
+
+2. Build the swift_library directly with a config.
```
Comment (brentleyjones) [inline doc/faq.md:24]:
```suggestion
2. Build the `swift_library` directly with a config.
```

File: doc/faq.md [hunk 11]
```diff
@@ -0,0 +1,39 @@
...
+
+You can alternately create a config to build for the desired platform in your .bazelrc:
```
Comment (brentleyjones) [inline doc/faq.md:26]:
```suggestion
You can alternately create a config to build for the desired platform in your `.bazelrc` file:
```

File: doc/faq.md [hunk 12]
```diff
@@ -0,0 +1,39 @@
...
+```
+
+And apply it when building:
+
+```
+bazel build //path/to/your/library --config=ios_simulator
+```
+
+The drawback to this approach is that the configs will likely differ between your application
+and swift library, resulting in the build cache not being shared.
```
Comment (brentleyjones) [inline doc/faq.md:39]:
```suggestion
and swift library, resulting in the caches not being shared.
```

File: doc/faq.md [hunk 13]
```diff
@@ -0,0 +1,39 @@
+# Frequently Asked Questions
+
+## How can I build a swift_library directly for iOS?
```
Comment (brentleyjones) [inline doc/faq.md:3]:
```suggestion
## How can I build a `swift_library` directly for iOS?
```

**Review by thii — APPROVED** [review]

**Pull request merged**